stack/app.py

```diff
@@ -73,22 +73,43 @@ def __init__(
             subnet_ids=[sb.subnet_id for sb in vpc.private_subnets],
         )
 
-        sg = ec2.SecurityGroup(self, f"{id}-cache-sg", vpc=vpc)
+        lambda_function_security_group = ec2.SecurityGroup(
+            self, f"{id}-lambda-sg", vpc=vpc
+        )
+        lambda_function_security_group.add_egress_rule(
+            ec2.Peer.any_ipv4(),
+            connection=ec2.Port(protocol=ec2.Protocol("ALL"), string_representation=""),
+            description="Allow lambda security group all outbound access",
+        )
+
+        cache_security_group = ec2.SecurityGroup(self, f"{id}-cache-sg", vpc=vpc)
+
+        cache_security_group.add_ingress_rule(
+            lambda_function_security_group,
+            connection=ec2.Port(protocol=ec2.Protocol("ALL"), string_representation=""),
+            description="Allow Lambda security group access to Cache security group",
+        )
+
         cache = escache.CfnCacheCluster(
             self,
             f"{id}-cache",
             cache_node_type=config.CACHE_NODE_TYPE,
             engine=config.CACHE_ENGINE,
             num_cache_nodes=config.CACHE_NODE_NUM,
-            vpc_security_group_ids=[sg.security_group_id],
+            vpc_security_group_ids=[cache_security_group.security_group_id],
             cache_subnet_group_name=sb_group.ref,
         )
 
-        vpc_access_policy_statement = iam.PolicyStatement(
+        vpc_log_access = iam.PolicyStatement(
             actions=[
                 "logs:CreateLogGroup",
                 "logs:CreateLogStream",
                 "logs:PutLogEvents",
+            ],
+            resources=["*"],
+        )
+        vpc_network_access = iam.PolicyStatement(
```

Comment:
@leothomas I think we can remove the `vpc_` prefix on both of these variable names since they are mostly unrelated to that as well. Not sure why the network permissions were needed initially (or even if they are now) but too nervous to remove